### ISSUE TITLE
Spoofed XFH header causes runtime error

### ISSUE THREAD

**Issue by stefan-guggisberg** [adobe/helix-html-pipeline#216]
**Description**
Spoofing an `X-Forwarded-Host` header with an invalid host name, e.g. 
```
X-Forwarded-Host: foo"><script>alert(313)</script><x=".com, main--myproj--myorg.hlx.page
```
causes a Runtime error: ` Invalid URL string`

**To Reproduce**
Curl the pipeline service with such a forged XFH header.

**Expected behavior**
A `400 Bad Request` response should be returned or the error should be handled gracefully.

**Comment by tripodsan** [comment]
this is a duplicate of #346... should be fixed.

**Comment by stefan-guggisberg** [comment]
Seems to be only occur in v6 and only on Cloudflare stack.

**Comment by stefan-guggisberg** [comment]
> this is a duplicate of #346... should be fixed.

on a node runtime, yes. on cloudflare, no

**Comment by stefan-guggisberg** [comment]
resolved via #217 and #218
